### ISSUE TITLE
Bug Report: Issues in polynomial_problem_generation.py and trainer.py

### ISSUE THREAD

**Issue by Ntimesp** [HiroshiKERA/calt-codebase#10]
## 1. `polynomial_problem_generation.py` (lines 165-167)

In [`scripts/dataset_generation/sagemath/polynomial_problem_generation.py`](https://github.com/HiroshiKERA/calt-codebase/blob/main/scripts/dataset_generation/sagemath/polynomial_problem_generation.py), lines 165-167:

```python
"sum_total_degree": sum(degrees),
"min_total_degree": min(degrees),
"max_total_degree": max(degrees),
```
The elements of degrees are of type Sage integer (sage.rings.integer.Integer) rather than Python int.
As a result, these fields may be filtered out in subsequent processing that expects Python integers, causing them not to appear in the final results.

## 2. `trainer.py` (line 69)
In [`src/calt/trainer/trainer.py`](https://github.com/HiroshiKERA/calt/blob/main/src/calt/trainer/trainer.py), line 69:

```python
predictions, labels = eval_preds
```
It seems that predictions are logits, not label IDs.
Therefore, the current computation of token_accuracy always returns 0, since it compares raw logits with label IDs.
